rpc-api/cmd/notify-rpc-api/grpc.go

```diff
@@ -0,0 +1,447 @@
+	Permissions       map[string][]string `json:"permissions"`
+}
+
+type Users []User
+
+func (user User) Protoify() *notify.User {
+	permissions := map[string]*notify.Permissions{}
+
+	for service, ps := range user.Permissions {
+		permissions[service] = &notify.Permissions{Permissions: ps}
+	}
+
+	return &notify.User{
+		Id:                user.Id,
+		Name:              user.Name,
+		AuthType:          user.AuthType,
+		EmailAddress:      user.EmailAddress,
+		FailedLoginCount:  user.FailedLoginCount,
+		PasswordChangedAt: user.PasswordChangedAt,
+		Permissions:       permissions,
+	}
+}
+
+func (u Users) Protoify() *notify.Users {
+	users := [](*notify.User){}
+
+	for _, user := range u {
+		users = append(users, user.Protoify())
+	}
+
+	return &notify.Users{Users: users}
+}
+
+func (s *server) GetUser(ctx context.Context, in *notify.ForUserRequest) (*notify.User, error) {
+	log.Printf("Getting user: [%s]\n", in.UserId)
+
+	userJSON, err := s.c.User(in.UserId)
+	if err != nil {
+		return nil, err
+	}
+
+	var user User
+	err = json.Unmarshal(userJSON, &user)
+	return user.Protoify(), err
+}
+
+func (s *server) GetUsers(ctx context.Context, in *notify.Empty) (*notify.Users, error) {
+	log.Println("Getting users")
+
+	usersJSON, err := s.c.Users()
+	if err != nil {
+		return nil, err
+	}
+
+	var users Users
+	err = json.Unmarshal(usersJSON, &users)
+	return users.Protoify(), err
+}
+
+func (s *server) GetService(ctx context.Context, in *notify.ServiceRequest) (*notify.Service, error) {
+	log.Printf("Getting service: [%s]\n", in.ServiceId)
+
+	serviceJSON, err := s.c.Service(in.ServiceId)
+	if err != nil {
+		return nil, err
+	}
+
+	var service Service
+	err = json.Unmarshal(serviceJSON, &service)
+	return service.Protoify(), err
+}
+
+func (s *server) ServicesForUser(ctx context.Context, in *notify.ForUserRequest) (*notify.Services, error) {
+	log.Printf("Getting services: [%s]\n", in.UserId)
+
+	servicesJSON, err := s.c.Services(in.UserId)
+	if err != nil {
+		return nil, err
+	}
+
+	var services Services
+	err = json.Unmarshal(servicesJSON, &services)
+	return services.Protoify(), err
+}
+
+func (s *server) TemplatesForService(ctx context.Context, in *notify.TemplatesRequest) (*notify.Templates, error) {
+	log.Printf("Getting templates: [%s]\n", in.ServiceId)
+
+	templatesJSON, err := s.c.Templates(in.ServiceId)
+	if err != nil {
+		return nil, err
+	}
+
+	var templates Templates
+	err = json.Unmarshal(templatesJSON, &templates)
+	return templates.Protoify(), err
+}
+
+func (s *server) CreateTemplate(ctx context.Context, in *notify.CreateTemplateRequest) (*notify.Template, error) {
+
+	templateJSON, err := s.c.CreateTemplate(
+		in.ServiceId,
+		Template{
+			Name:         in.Name,
+			TemplateType: "email",
+			Subject:      in.Subject,
+			Content:      in.Content,
+			Service:      in.ServiceId,
+			CreatedBy:    in.CreatedBy,
+			CreatedAt:    time.Now().Format(time.RFC3339),
+			ProcessType:  "normal",
```

Comment:
Does this need to be extracted to a const somewhere?

Reply:
this is prototype code